src/Core/Models/Domain/Account.cs

```diff
@@ -117,9 +117,14 @@ public AccountSettings(AccountSettings copy)
 
         public class AccountVolatileData
         {
+            public UserKey UserKey;
+            public MasterKey MasterKey;
+            public EncString UserKeyPinEphemeral;
+            public bool? BiometricLocked;
+            [Obsolete("Jul 6 2023: Key has been deprecated. We will use the User Key in the future. It remains here for migration during app upgrade.")]
             public SymmetricCryptoKey Key;
+            [Obsolete("Jul 6 2023: PinProtectedKey has been deprecated in favor of UserKeyPinEphemeral. It remains here for migration during app upgrade.")]
             public EncString PinProtectedKey;
-            public bool? BiometricLocked;
```

Comment:
The `Key` property is needed because if the account is upgraded and the vault timeout is set to `never`, we could potentially have a situation where the new `MasterKey` is never placed on unlock/login. The old key is still retained in memory though, so we can check for it and upgrade it then:
```c#
        public async Task<MasterKey> GetMasterKeyAsync(string userId = null)
        {
            var masterKey = await _stateService.GetMasterKeyAsync(userId);
            if (masterKey == null)
            {
                // Migration support
                masterKey = new MasterKey((await _stateService.GetKeyDecryptedAsync(userId)).Key);
                if (masterKey != null)
                {
                    await SetMasterKeyAsync(masterKey, userId);
                }
            }
            return masterKey;
        }
```

The `PinProtectedKey` is used during the Pin migration, which can only happen once the user unlocks with their pin.